frame/treasury/src/lib.rs

```diff
@@ -374,34 +349,45 @@ impl<T: Trait> Module<T> {
 	}
 
 	// Spend some money!
-	fn spend_funds() {
-		let mut budget_remaining = Self::pot();
-		Self::deposit_event(RawEvent::Spending(budget_remaining));
+	fn spend_funds<C: Currency<T::AccountId>>() {
+		let mut budget_remaining = Self::pot::<T::RingCurrency>();
+		Self::deposit_event(RawEvent::Spending(StakingBalance::RingBalance(budget_remaining)));
```

Comment:
Here is the problem needs to be fixed currently. 

It's important to infer the passing `Currency` to spend different currency, but `Currency` is only a `trait` which doesn't have `Sized` and can .not make into an object(for inferring different passing type).

Reply:
Solved this by a common `enum` parameter...